lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -2089,6 +2101,72 @@ def _enable_ssh_on_windows(self, node: Node) -> None:
         log.debug("SSH script result:")
         log.dump_json(logging.DEBUG, result)
 
+    def _get_vhd_os_disk_size(self, blob_url: str) -> int:
+        vhd_blob = BlobClient.from_blob_url(
+            blob_url=blob_url,
+            credential=self.credential,
+        )
+        properties = vhd_blob.get_blob_properties()
+        assert properties.size
+        return int(properties.size / 1024 / 1024 / 1024)
+
+    def _get_sig_info(
+        self, shared_image: SharedImageGallerySchema
+    ) -> GalleryImageVersion:
+        compute_client = get_compute_client(self)
+        gallery_images = compute_client.gallery_image_versions.list_by_gallery_image(
+            resource_group_name=shared_image.resource_group_name,
+            gallery_name=shared_image.image_gallery,
+            gallery_image_name=shared_image.image_definition,
+        )
+        if shared_image.image_version.lower() != "latest":
+            found_image = compute_client.gallery_image_versions.get(
+                resource_group_name=shared_image.resource_group_name,
+                gallery_name=shared_image.image_gallery,
+                gallery_image_name=shared_image.image_definition,
+                gallery_image_version_name=shared_image.image_version,
+                expand="ReplicationStatus",
+            )
+        else:
+            image: GalleryImageVersion = None
+            time: Optional[datetime] = None
+            for image in gallery_images:
+                gallery_image = compute_client.gallery_image_versions.get(
+                    resource_group_name=shared_image.resource_group_name,
+                    gallery_name=shared_image.image_gallery,
+                    gallery_image_name=shared_image.image_definition,
+                    gallery_image_version_name=image.name,
+                    expand="ReplicationStatus",
+                )
+                if not time:
+                    time = gallery_image.publishing_profile.published_date
+                if gallery_image.publishing_profile.published_date > time:
+                    time = gallery_image.publishing_profile.published_date
+                    found_image = gallery_image
+        return found_image
+
+    def _get_sig_size(self, shared_image: SharedImageGallerySchema) -> int:
```

Comment:
As vhd, it should be like `_get_sig_os_disk_size`

Reply:
updated